### PR TITLE
annotations: test that from __future__ import annotations works

### PR DESCRIPTION
**Pull request by JelleZijlstra**

### CODE REVIEW

File: pyanalyze/test_annotations.py [hunk 1]
```diff
@@ -404,6 +404,19 @@ def capybara(x: Pattern[str]):
 """
         )
 
+    @skip_before((3, 7))
+    def test_future_annotations(self):
+        self.assert_passes(
+            """
+from __future__ import annotations
+from typing import List
```
Comment (dimaqq) [inline pyanalyze/test_annotations.py:412]: Does this test that unused future annotations work?
Or was the intention to use some time without quotes or forward declaration?

Comment (JelleZijlstra) [reply pyanalyze/test_annotations.py:412]: It tests that even with stringified annotations, pyanalyze understands the annotations.

Comment (dimaqq) [reply pyanalyze/test_annotations.py:412]: Sorry I guess I don't really know how pyanalyze works...
I don't see a stringified annotation verbatim in the snippet 🤷

Comment (JelleZijlstra) [reply pyanalyze/test_annotations.py:412]: `from __future__ with annotations` is what stringifies the annotations.

Comment (dimaqq) [reply pyanalyze/test_annotations.py:412]: ahhh, ok, I guess I was just confused that the code snippet "works" same with or without the future import, and failed to realise that typing bits are subtly different.

**Pull request merged**